src/sources/kubernetes/mod.rs

```diff
@@ -105,48 +119,161 @@ impl TimeFilter {
 fn file_source(
     kube_name: &str,
     globals: &GlobalOptions,
+    config: &KubernetesConfig,
 ) -> crate::Result<(mpsc::Receiver<Event>, Source)> {
-    let mut config = FileConfig::default();
-
-    // TODO: Add exclude_namspace option, and with it, in config, exclude kube-system namespace.
-    // This is correct, but on best effort basis filtering out of logs from kuberentes system components.
-    // More specificly, it will work for all Kubernetes 1.14 and higher, and for some bellow that.
-    config
-        .exclude
-        .push((LOG_DIRECTORY.to_owned() + r"kube-system_*").into());
+    let mut file_config = FileConfig::default();
 
-    // TODO: Add exclude_namspace option, and with it, in config, exclude namespace used by vector.
-    // NOTE: for now exclude images with name vector, it's a rough solution, but necessary for now
-    config
-        .exclude
-        .push((LOG_DIRECTORY.to_owned() + r"*/vector*").into());
+    file_source_include(config, &mut file_config)?;
+    file_source_exclude(config, &mut file_config);
 
-    config
-        .include
-        .push((LOG_DIRECTORY.to_owned() + r"*/*/*.log").into());
-
-    config.start_at_beginning = true;
+    file_config.start_at_beginning = true;
 
     // Filter out files that certainly don't have logs newer than now timestamp.
-    config.ignore_older = Some(10);
+    file_config.ignore_older = Some(10);
 
     // oldest_first false, having all pods equaly serviced is of greater importance
     //                     than having time order guarantee.
 
     // CRI standard ensures unique naming.
-    config.fingerprinting = FingerprintingConfig::DevInode;
+    file_config.fingerprinting = FingerprintingConfig::DevInode;
 
     // Have a subdirectory for this source to avoid collision of naming its file source.
-    config.data_dir = Some(globals.resolve_and_make_data_subdir(None, kube_name)?);
+    file_config.data_dir = Some(globals.resolve_and_make_data_subdir(None, kube_name)?);
 
     let (file_send, file_recv) = mpsc::channel(1000);
-    let file_source = config
+    let file_source = file_config
         .build("file_source", globals, file_send)
         .map_err(|e| format!("Failed in creating file source with error: {:?}", e))?;
 
     Ok((file_recv, file_source))
 }
 
+fn file_source_include(
+    config: &KubernetesConfig,
+    file_config: &mut FileConfig,
+) -> crate::Result<()> {
+    // Default includes
+    let empty = ["".to_string()];
+    let any = ["*".to_string()];
+    let uid_glob = to_uid_forms("");
+
+    // Cross product of includes
+    for object_uid in flatten_include(
```

Comment:
I am having a bit trouble following these three for loops, is there a way we can improve this a bit?